instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/internal/ParameterizedClass.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.internal;
+
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.lang.reflect.TypeVariable;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Helper class for reflecting over the type hierarchy of parameterized types where the generic type
+ * arguments are reified and/or mapped to the type parameters of the implemented superclasses or
+ * interfaces.
+ */
+public final class ParameterizedClass implements ParameterizedType {
```

Comment:
Is it necessary to have this class implement `ParameterizedType`? I think there's a lot of logic around owner types that's only used in the `getTypeName()` - which is only used in tests.